php_phongo.c

```diff
@@ -1963,22 +1963,64 @@ PHP_GSHUTDOWN_FUNCTION(mongodb)
 PHP_MINFO_FUNCTION(mongodb)
 {
 	php_info_print_table_start();
-	php_info_print_table_header(2, "mongodb support", "enabled");
-	php_info_print_table_row(2, "mongodb version", MONGODB_VERSION_S);
-	php_info_print_table_row(2, "mongodb stability", MONGODB_STABILITY_S);
+	php_info_print_table_header(2, "MongoDB support", "enabled");
+	php_info_print_table_row(2, "MongoDB extension version", MONGODB_VERSION_S);
+	php_info_print_table_row(2, "MongoDB extension stability", MONGODB_STABILITY_S);
+
+#ifdef HAVE_SYSTEM_LIBBSON
+	php_info_print_table_row(2, "libbson headers version", BSON_VERSION_S);
+	php_info_print_table_row(2, "libbson library version", bson_get_version());
+#else
+	php_info_print_table_row(2, "libbson bundled version", BSON_VERSION_S);
+#endif
+
 #ifdef HAVE_SYSTEM_LIBMONGOC
 	php_info_print_table_row(2, "libmongoc headers version", MONGOC_VERSION_S);
 	php_info_print_table_row(2, "libmongoc library version", mongoc_get_version());
 #else
 	/* Bundled libraries, buildtime = runtime */
-	php_info_print_table_row(2, "libmongoc version", MONGOC_VERSION_S);
+	php_info_print_table_row(2, "libmongoc bundled version", MONGOC_VERSION_S);
 #endif
-#ifdef HAVE_SYSTEM_LIBBSON
-	php_info_print_table_row(2, "libbson headers version", BSON_VERSION_S);
-	php_info_print_table_row(2, "libbson library version", bson_get_version());
+
+#ifdef MONGOC_ENABLE_SSL
+	php_info_print_table_row(2, "libmongoc SSL", "enabled");
+# if defined(MONGOC_ENABLE_SSL_OPENSSL)
+	php_info_print_table_row(2, "libmongoc SSL library", "OpenSSL");
+# elif defined(MONGOC_ENABLE_SSL_LIBRESSL)
+	php_info_print_table_row(2, "libmongoc SSL library", "LibreSSL");
+# elif defined(MONGOC_ENABLE_SSL_SECURE_TRANSPORT)
+	php_info_print_table_row(2, "libmongoc SSL library", "Secure Transport");
+# elif defined(MONGOC_ENABLE_SSL_SECURE_CHANNEL)
+	php_info_print_table_row(2, "libmongoc SSL library", "Secure Channel");
+# endif
+#else
+	php_info_print_table_row(2, "libmongoc SSL", "disabled");
+#endif
+
+#ifdef MONGOC_ENABLE_CRYPTO
+	php_info_print_table_row(2, "libmongoc crypto", "enabled");
+# if defined(MONGOC_ENABLE_CRYPTO_LIBCRYPTO)
+	php_info_print_table_row(2, "libmongoc crypto library", "libcrypto");
+# elif defined(MONGOC_ENABLE_CRYPTO_COMMON_CRYPTO)
+	php_info_print_table_row(2, "libmongoc crypto library", "Common Crypto");
+# elif defined(MONGOC_ENABLE_CRYPTO_CNG)
+	php_info_print_table_row(2, "libmongoc crypto library", "CNG");
+# endif
+# ifdef MONGOC_ENABLE_CRYPTO_SYSTEM_PROFILE
+	php_info_print_table_row(2, "libmongoc crypto system profile", "enabled");
+# else
+	php_info_print_table_row(2, "libmongoc crypto system profile", "disabled");
```

Comment:
AFAIK, this is currently only relevant to OpenSSL, which uses libcrypto. In any event, I liked including it at all times (if crypto is enabled) for consistency.

@bjori: Am I correct in assuming this will never apply to LibreSSL, despite the fact that it also uses libcrypto?